velox/functions/lib/MysqlDateTime.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/MysqlDateTime.h"
+#include <string>
+#include "velox/functions/lib/DateTimeFormatterBuilder.h"
+
+namespace facebook::velox::functions {
+
+MysqlDateTime::MysqlDateTime(const std::string& format) {
+  DateTimeFormatterBuilder builder(format.size());
+  if (format.empty()) {
+    VELOX_USER_FAIL("Invalid pattern specification.");
+  }
```

Comment:
Probably not, for the builder the input should be buffer size (sorry for the naming, should have changed formatSize to literalBufferSize). If the user knows there is no literals in the format then it can give 0 to the builder constructor and still be a valid builder.